data_preparation/row_selection.py

```diff
@@ -64,7 +64,7 @@ def process(self, inputs):
             sys.exit(-1)
 
         data_indexes = data.data.index.values.tolist()
-        data.inputs_split["training"] = data_indexes.copy()
-        data.inputs_split["all"] = data_indexes.copy()
+        inputs.inputs_split["training"] = data_indexes.copy()
+        inputs.inputs_split["all"] = data_indexes.copy()
 
-        return data
+        return inputs
```

Comment:
I'm not sure I understand what you're doing here. You are removing rows from `data`, but then you return a modified `inputs` which still includes the rows that were supposed to be removed. By the way, since Python passes function arguments by reference and not by value, you are changing the original object that was passed as `inputs`.

If you're just trying to change an independent copy of `inputs` to return that, the addition of `copy()` at line 51 should be enough. Indeed, we were changing `data` directly before.

Reply:
The idea is to keep the old implementation to retrieve the indexes that will not be considered going forward: since actually dropping them is not consistent with the rest of the code (thus the out-of-bounds error), RowSelection now returns the same inputs object and just updates the inputs_split dictionary

Reply:
First of all, I would not change the `inputs` argument directly. As mentioned, it can be dangerous. Then, if we only keep track of indexes, why not just create a list of valid indexes and update that instead of removing rows from a DataFrame, which looks way more expensive?

Reply:
Most of the data_preparation files have the "data = inputs" line at the beginning, thus I thought it was intended. Should I change it in all of them? E.g. column_selection.py, rename_column.py, ecc. Anyways, now should be fixed, at least in SkipRows

Reply:
I don't know if it was intended, but it is certainly unnecessary here. It is probably a leftover from copy-pasting. Also, when I implemented the `RowSelection` class, I did not know about the argument by reference quirk (nor did the original author of these modules, possibly)